modin/data_management/factories/dispatcher.py

```diff
@@ -35,32 +62,48 @@ def stub(*args, **kw):
 
 class StubFactory(factories.BaseFactory):
     """
-    A factory that does nothing more than raise NotImplementedError when any method is called.
+    Factory that does nothing more than raise NotImplementedError when any method is called.
+
+    Notes
+    -----
     Used for testing purposes.
     """
 
     io_cls = StubIoEngine()
 
     @classmethod
     def set_failing_name(cls, factory_name):
+        """
+        Fill in `.io_cls` class attribute with ``StubIoEngine`` engine.
+
+        Parameters
+        ----------
+        factory_name : str
+            Name to pass to the ``StubIoEngine`` constructor.
+        """
         cls.io_cls = StubIoEngine(factory_name)
         return cls
 
 
 class EngineDispatcher(object):
     """
-    This is the 'ingestion' point which knows where to route the work
+    Class that routes IO-work to the engines.
+
+    This class is responsible for keeping selected engine up-to-date and dispatching
+    calls of IO-functions to its actual engine-specific implementations.
     """
 
     __engine: factories.BaseFactory = None
 
     @classmethod
     def get_engine(cls) -> factories.BaseFactory:
+        """Get current execution engine."""
```

Comment:
We should rename `cls.__engine` to `cls.__factory` (and all related stuff) in a separate PR.